cmd/talosctl/cmd/talos/root.go

```diff
@@ -94,6 +97,15 @@ func WithClient(action func(context.Context, *client.Client) error) error {
 				return fmt.Errorf("nodes are not set for the command: please use `--nodes` flag or configuration file to set the nodes to run the command against")
 			}
 
+			if Cluster == "" {
+				configContext := c.GetConfigContext()
+				if configContext == nil {
+					return configContextErr
+				}
+
+				Cluster = configContext.Cluster
```

Comment:
this `Cluster` varialbe if set should go to `client.New` as an option: `WithCluster(Cluster)`.